flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/process/ProcessTableOperator.java

```diff
@@ -331,12 +361,24 @@ private void setStateDescriptors() {
         this.stateDescriptors = stateDescriptors;
     }
 
-    @SuppressWarnings("unchecked")
     private void setStateHandles() {
         final KeyedStateStore keyedStateStore = getKeyedStateStore();
-        final ValueState<RowData>[] stateHandles = new ValueState[stateDescriptors.length];
+        final State[] stateHandles = new State[stateDescriptors.length];
         for (int i = 0; i < stateInfos.size(); i++) {
-            stateHandles[i] = keyedStateStore.getState(stateDescriptors[i]);
```

Comment:
I am not understanding this for loop. I wonder if the comment might help.
- We loop on `stateInfos.size()` not does not refer to `stateInfos[i]` in the loop. Would this be better :  `for (StateDescriptor stateDescriptor: stateDescriptors) {`

Reply:
the size of stateInfos, stateDescriptors, and stateHandles is equal. But I agree that this comment is valid.